File: leetcode_problems/cpp/lc203_remove_linked_list_elements.cpp
struct ListNode
{
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

class Solution
{
public:
    ListNode *removeElements(ListNode *head, int val)
    {
        ListNode dummy(0);
        dummy.next = head;
        ListNode *node = &dummy;

        while (node->next != nullptr)
        {
            if (node->next->val == val)
            {
                ListNode *temp = node->next;
                node->next = node->next->next;
                delete temp;
            }
            else
            {
                node = node->next;
            }
        }

        return dummy.next;
    }
};

File: leetcode_problems/typescript/src/lc203_remove_linked_list_elements.ts
function removeElements(head: ListNode | null, val: number): ListNode | null {
  let dummy = ListNode(0, head);
  let node = dummy;
  while (node.next !== null) {
    if (node.next.val === val) {
      node.next = node.next.next;
    } else {
      node = node.next;
    }
  }
  return dummy.next;
}
